plugins/source/aws/codegen/recipes/base.go

```diff
@@ -418,3 +424,28 @@ func CreateReplaceTransformer(replace map[string]string) func(field reflect.Stru
 		return name, nil
 	}
 }
+
+func validateServiceMultiplex(multiplexerCall string) error {
+	re := regexp.MustCompile(`\"(.*?)\"`)
+	// Find the value of the service parameter
+	submatchAll := re.FindStringSubmatch(multiplexerCall)
+	if len(submatchAll) != 2 {
+		return nil
+	}
+
+	t := client.ReadSupportedServiceRegions()
+	services := make(map[string]bool)
+
+	for _, partition := range t.Partitions {
+		for service := range partition.Services {
+			if _, ok := services[service]; !ok {
+				services[service] = true
+			}
+		}
+	}
+	if _, ok := services[submatchAll[1]]; !ok {
+		return fmt.Errorf("invalid partition: %s", submatchAll[1])
+	}
+	return nil
+
```

Comment:
nit: rogue newline (or move L449 to here and have the newline after the if block)